splitio/engine/impressions/adapters.py

```diff
@@ -119,14 +119,82 @@ def _expire_keys(self, queue_key, key_default_ttl, total_keys, inserted):
         if total_keys == inserted:
             self._redis_client.expire(queue_key, key_default_ttl)
 
-    def _uniques_formatter(self, uniques):
+class PluggableSenderAdapter(ImpressionsSenderAdapter):
+    """In Memory Impressions Sender Adapter class."""
+
+    MTK_QUEUE_KEY = 'SPLITIO.uniquekeys'
+    MTK_KEY_DEFAULT_TTL = 3600
+    IMP_COUNT_QUEUE_KEY = 'SPLITIO.impressions.count'
+    IMP_COUNT_KEY_DEFAULT_TTL = 3600
+
+    def __init__(self, adapter_client, prefix=None):
         """
-        Format the unique keys dictionary array to a JSON body
+        Initialize pluggable sender adapter instance
+
+        :param telemtry_http_client: instance of telemetry http api
+        :type telemtry_http_client: splitio.api.telemetry.TelemetryAPI
+        """
+        self._adapter_client = adapter_client
+        self._prefix = ""
+        if prefix is not None:
+            self._prefix = prefix + "."
+
+    def record_unique_keys(self, uniques):
+        """
+        post the unique keys to storage.
 
         :param uniques: unique keys disctionary
         :type uniques: Dictionary {'feature1': set(), 'feature2': set(), .. }
+        """
+        bulk_mtks = _uniques_formatter(uniques)
+        try:
+            inserted = self._adapter_client.push_items(self.MTK_QUEUE_KEY, *bulk_mtks)
+            self._expire_keys(self._prefix + self.MTK_QUEUE_KEY, self.MTK_KEY_DEFAULT_TTL, inserted, len(bulk_mtks))
+            return True
+        except RedisAdapterException:
+            _LOGGER.error('Something went wrong when trying to add mtks to storage adapter')
+            _LOGGER.error('Error: ', exc_info=True)
+            return False
 
-        :return: unique keys JSON array
-        :rtype: json
+    def flush_counters(self, to_send):
         """
-        return [json.dumps({'f': feature, 'ks': list(keys)}) for feature, keys in uniques.items()]
+        post the impression counters to storage.
+
+        :param to_send: unique keys disctionary
+        :type to_send: Dictionary {'feature1': set(), 'feature2': set(), .. }
+        """
+        try:
+            resulted = 0
+            for pf_count in to_send:
+                resulted = self._adapter_client.increment(self._prefix + self.IMP_COUNT_QUEUE_KEY + "." + pf_count.feature + "::" + str(pf_count.timeframe), pf_count.count)
```

Comment:
`self._prefix + self.IMP_COUNT_QUEUE_KEY + "." + pf_count.feature + "::" + str(pf_count.timeframe)` can be calculated and stored into a variable, to avoid recalculating it twice.